.github/workflows/testing.yml

```diff
@@ -60,14 +60,51 @@ jobs:
           cp .env.testing app/.env
           cd app/
           mkdir -p static_files
-          python manage.py test
+          python manage.py test --exclude-tag=selenium
         env:
           DJANGO_SETTINGS_MODULE: app.settings
           DATABASE_URL: postgres://sadilar:sadilar@localhost:5432/test_db
       - name: Manager Check
         run: |
           cd app/
           python manage.py check
+  selenium:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:16
+        env:
+          POSTGRES_USER: sadilar
+          POSTGRES_PASSWORD: sadilar
+          POSTGRES_DB: test_db_1
+        ports:
+          - 5432:5432
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+      - name: Install Dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements-test.txt
+          sudo apt-get install -y gettext
+      - name: Create logging folder
+        run: |
+          sudo  mkdir -p /logging
+          sudo chown runner:runner /logging
+      - name: Run Selenium Tests
+        run: |
+          cp .env.testing app/.env
+          cd app/
+          mkdir -p static_files
+          python manage.py test --tag=selenium
+        env:
+          DJANGO_SETTINGS_MODULE: app.settings
+          DATABASE_URL: postgres://sadilar:sadilar@localhost:5432/test_db
```

Comment:
The repetition is a bit unfortunate. One solution is to combine everything before the last step and label with a yaml anchor or two to allow reuse, but that takes us towards the parts of yaml that is less readable and probably not as well known.

Another possibility is to combine all the consecutive shell commands (`pip install` up to `mkdir -p static_files`) into one step. See

Reply:
I very much agree about the repetition. Unfortunately, I haven't been able to find a better way to do it yet. It looks like yaml anchors are unfortunately not supported by GitHub actions: https://github.com/actions/runner/issues/1182